src/python/cli_new/lib/cli/http.py

```diff
@@ -41,6 +41,8 @@ def read_endpoint(addr, endpoint):
 
     try:
         url = "{addr}/{endpoint}".format(addr=addr, endpoint=endpoint)
+        if query is not None:
+          url += "?{query}".format(query=urllib.parse.urlencode(query))
```

Comment:
Query parameters being common in the HTTP endpoints, having the ability to add some when reading an endpoint makes sense ✅.